### DISCUSSION TITLE
Cannot find module 'payload/node' in latest beta version

### DISCUSSION THREAD

**Discussion by creonico** [payloadcms/payload#7512] [Q&A]
# Documentation Issue

I am trying to use payloadcms outside of nextjs as described in this doc https://payloadcms.com/docs/beta/local-api/outside-nextjs, but I get the following error when trying to import importConfig  from payload/node:
```
server.ts:2:30 - error TS2307: Cannot find module 'payload/node' or its corresponding type declarations.
```

Here is my server.ts script, i run this in node with `ts-node server.ts`:
```typescript
// server.ts
import { getPayload } from 'payload';
import { importConfig } from 'payload/node';
```
The importConfig is not even used in this example from the doc, but the line `const payload = await getPayload({ config })` uses a variable config that is never declared in this example.

In this doc, however, https://payloadcms.com/docs/beta/local-api/overview#importing-it, it does show how to use the importConfig function, so I tried this method but that is where I get the error above.

## Additional Details

I tried to running the same script in deno with `deno run server.ts`, but I get this error:
```
Task seed deno run server.ts
error: Uncaught SyntaxError: The requested module 'node:module' does not provide an export named 'register'
    at <anonymous> (file:.../deno/npm/registry.npmjs.org/payload/3.0.0-beta.73/dist/utilities/importWithoutClientFiles.js:1:10)
```

**Comment by jmikrut** [comment]
Hey @creonico there may be a few different things going on here.

1. we have not tested this with `deno` yet, but at some point we will
2. you need to make sure that you have a Node version that is compatible with Payload (Node.js version 20.9.0+)
3. Payload is an ESM package now, so you'll want to make sure that your tsconfig is set up properly to be able to handle ESM. If you are in a CommonJS environment, you will not be able to use the `importConfig` function unless you explicitly name / mark your script as ESM rather than CommonJS.

You can find an example of how to use all of this in the 3.0 Demo:
https://github.com/payloadcms/payload-3.0-demo/blob/main/src/scripts/standalone-script.js

Check out your Node version - maybe that will fix your issue. I'll convert this to a discussion though so we can continue to help accordingly.

**Reply by creonico** [reply]
I got the node version working now by adjusting the tsconfig to handle ESM. But what about the deno version? I understand that it has not been tested yet but shouldn't it work out of the box? The error says that `The requested module 'node:module' does not provide an export named 'register'` but the node:module does export register https://nodejs.org/docs/latest-v20.x/api/module.html#moduleregisterspecifier-parenturl-options. Is this an issue in deno's end? My current nodejs version is v20.12.2

**Reply by denolfe** [reply]
According to the Node.js docs, that version should have `register` available. I'd guess this is a Deno issue.
